src/eth/rpc.ts

```diff
@@ -6,10 +6,11 @@ const fetchSubgraph = createApolloFetch({
     uri: `${process.env.SUBGRAPH_URL}`,
   });
 
-export const getEarnings = async(address:string, endRound: BigNumber): Promise<{delegator: string, pendingStake: BigNumber, pendingFees: BigNumber}> => {
+export const getEarnings = async(address:string, endRound: BigNumber, delegateAddr?: string): Promise<{delegator: string, delegate: string, pendingStake: BigNumber, pendingFees: BigNumber}> => {
     try {
-        const earnings: {delegator: string, pendingStake: BigNumber, pendingFees: BigNumber} = {
+        const earnings: {delegator: string, delegate: string, pendingStake: BigNumber, pendingFees: BigNumber} = {
             delegator: address,
+            delegate: delegateAddr || "",
```

Comment:
`getEarnings()` is also used by `earnings()` but it does not use _delegate_ 
Hence, delegateAddr is an optional param and an empty string is returned when called without it.

Reply:
Instead of passing the delegate as a param here just to include it in the returned object, could the caller be responsible for composing the result of this function into another object that includes the delegate as a field? Then, we wouldn't need to handle this case.

See my other comment in `getDelegators()` for an example of how we could do this.